src/Components/Web/src/Forms/InputBase.cs

```diff
@@ -74,7 +76,18 @@ public abstract class InputBase<TValue> : ComponentBase, IDisposable
             var hasChanged = !EqualityComparer<TValue>.Default.Equals(value, Value);
             if (hasChanged)
             {
+                _parsingFailed = false;
+
+                // If we don't do this, then when the user edits from A to B, we'd:
+                // - Do a render that changes back to A
+                // - Then send the updated value to the parent, which sends the B back to this component
+                // - Do another render that changes it to B again
+                // The unnecessary reversion from B to A can cause selection to be lost while typing
+                // A better solution would be somehow forcing the parent component's render to occur first,
+                // but that would involve a complex change in the renderer to keep the render queue sorted
+                // by component depth or similar.
                 Value = value;
+
                 _ = ValueChanged.InvokeAsync(Value);
```

Comment:
Side note since you mentioned https://github.com/dotnet/aspnetcore/issues/44105

Do you think it's reasonable to move the logic for invoking ValueChanged as well as `EditContext.NotifyFieldChanged` to the event handler? (Essentially manually hand-coding `@bind:after` for the `input` element binding)

That was my original idea/direction for the issue, but it is breaking-ish. That said, since we are doing a somewhat breaking change here, I think it would be a good idea to tackle both things at the same time (not in this PR) if my suggestion is something we want to follow.

Reply:
That sounds like a similar suggestion to the one raised at https://github.com/dotnet/aspnetcore/pull/46434#issuecomment-1415959720

I'm definitely open to doing that but (as per [this](https://github.com/dotnet/aspnetcore/pull/46434#issuecomment-1416085924)) would rather keep it separate.

Reply:
> Essentially manually hand-coding @bind:after for the input element binding

Ideally we might even be able to just use @bind:get/@bind:set and change the Input* components to be written as `.razor` files. I started doing that but it derailed the PR and introduced too many independent changes so I think we should tackle it separately.

Reply:
@SteveSandersonMS how would one implement the equivalent of SetUpdatesAttributeName when using .razor files? I just discovered that method today why investigating an [issue](https://forum.radzen.com/t/clearing-a-textbox-after-valuechange/17323/8). It turns out this is what I need but I don't know how to use SetUpdatesAttributeName in a [razor file](https://github.com/radzenhq/radzen-blazor/blob/master/Radzen.Blazor/RadzenTextBox.razor). Is there any way or should I abandon the razor file and use BuildRenderTree?

Reply:
AFAIK, `<html-element @bind-Value="..." />` in Razor generates the `SetUpdatesAttributeName()` as part of generated `BuildRenderTree()` method.

Reply:
Hi @hakenr! I will test of this addresses the issue.